controllers/predicate/predicate.go

```diff
@@ -205,3 +189,33 @@ func IsSnapshotLease() predicate.Predicate {
 		},
```

Comment:
Why do we need to react to delete event of a snapshot lease?

Reply:
Is it not a good idea to check the revision diff for one last time and save snapshot using compaction?

Reply:
In which even will a lease be deleted?

Reply:
added changes to not react to delete event